src/core/issues-notes.js

```diff
@@ -147,59 +175,79 @@ function handleIssues($ins, ghIssues, conf) {
   }
 }
 
-async function fetchIssuesFromGithub({ githubAPI }) {
-  const issues = [];
-  const issueNumbers = [...document.querySelectorAll(".issue[data-number]")]
+async function fetchAndStoreGithubIssues(githubAPI) {
+  const specIssues = document.querySelectorAll(".issue[data-number]");
+  if (specIssues.length > MAX_GITHUB_REQUESTS) {
+    const msg =
+      `Your spec contains ${specIssues.length} Github issues, ` +
+      `but GitHub only allows ${MAX_GITHUB_REQUESTS} requests. Some issues might not show up.`;
+    pub("warning", msg);
+  }
+  const issuePromises = [...specIssues]
     .map(elem => Number.parseInt(elem.dataset.number, 10))
-    .filter(number => number);
-  for (const issueNumber of issueNumbers) {
-    let issue; // depends on how fetching goes...
-    try {
+    .filter(issueNumber => issueNumber)
+    .map(async issueNumber => {
       const issueURL = `${githubAPI}/issues/${issueNumber}`;
-      const response = await fetch(issueURL, {
-        // Get back HTML content instead of markdown
-        // See: https://developer.github.com/v3/media/
-        Accept: "application/vnd.github.v3.html+json",
+      const request = new Request(issueURL, {
+        headers: {
+          // Get back HTML content instead of markdown
+          // See: https://developer.github.com/v3/media/
+          Accept: "application/vnd.github.v3.html+json",
+        },
       });
-      if (!response.ok) {
-        switch (response.status) {
-          case 404:
-            throw new Error(
-              "Couldn't find issue on Github. Check if it exist?"
-            );
-          default:
-            throw new Error(
-              "Network error. Github is down? or too many requests?"
-            );
-        }
-      }
-      issue = await response.json(); // can throw too
-    } catch (err) {
-      console.error(err);
-      const msg = `Error fetching issue #${issueNumber} from GitHub. ${
-        err.message
-      }. See developer console.`;
-      pub("error", msg);
-      issue = { title: "", number: issueNumber, state: "" };
-    }
-    issues.push([issueNumber, issue]);
+      const response = await fetchAndCache(request);
```

Comment:
@saschanaz, ok 404s etc, are now passed on. Exceptions go unhandled. Should we do anything else here?